boot/zephyr/os.c

```diff
@@ -24,29 +24,21 @@
 
 #define MBEDTLS_CONFIG_FILE CONFIG_MBEDTLS_CFG_FILE
 #include <mbedtls/platform.h>
+#include <mbedtls/memory_buffer_alloc.h>
 
-/* D(void *os_malloc(size_t size)) */
-void *os_calloc(size_t nelem, size_t size)
-{
-    /* Note that this doesn't check for overflow.  Assume the
-     * calls only come from within the app. */
-    size_t total = nelem * size;
-    void *buf = k_malloc(total);
-    if (buf) {
-        memset(buf, 0, total);
-    }
-    return buf;
-}
+/*
+ * This is the heap for mbed TLS.  The value needed depends on the key
+ * size and algorithm used.  For RSA-2048, 6144 bytes seems to be
```

Comment:
Looks good to me, but how was this tested? Did you run STACK_ANALYZE() in zephyr or something like that?

Reply:
I couldn't get any of the heap analysis code in mbed TLS to work (it didn't even compile, and didn't work once I fixed those problems). I ended up doing kind of a binary search, and being a bit generous.